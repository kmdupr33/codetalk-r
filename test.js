const nearley = require("nearley");
const grammar = require("./grammar");
const fs = require("fs");

describe("codetalk", () => {
  // Helpers
  let parser;
  const readFile = (path) => fs.readFileSync(path, 'utf8');
  const expectToParse = (parser, input) => {
    parser.feed(input);
    expect(parser.results.length).toBe(1);
    expect(parser.results).toMatchSnapshot();
  }
  // ---
  
  beforeEach(() => {    
    parser = new nearley.Parser(nearley.Grammar.fromCompiled(grammar), {
      keepHistory: true
    });    
  });
  test("hello_world.codetalk parses correctly", () => {      
    const input = readFile('./code_talks/hello_world.codetalk');        
    expectToParse(parser, input);
  });  
  test("basic.codetalk parses correctly", () => {
    const input = readFile('./code_talks/basic.codetalk');
    expectToParse(parser, input);
  })  
  test("complex.codetalk parses correctly", () => {
    const input = readFile('./code_talks/complex.codetalk');
    expectToParse(parser, input);
  })
  test("paragraphs.codetalk parses correctly", () => {
    const input = readFile('./code_talks/paragraphs.codetalk');
    expectToParse(parser, input);
  })
});
