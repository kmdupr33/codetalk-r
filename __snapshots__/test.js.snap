// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`codetalk basic.codetalk parses correctly 1`] = `
Array [
  Array [
    "codetalk",
    Array [
      "slide",
      Array [
        Array [
          Array [
            "#",
            "hello world",
          ],
          Array [],
        ],
        Array [
          Array [
            "##",
            " goodbye",
          ],
          Array [],
        ],
        Array [
          "idk why you say goodbye",
          Array [],
        ],
      ],
      "end",
    ],
    Array [
      "slide",
      Array [
        Array [
          "goodbye world",
          Array [],
        ],
      ],
      "end",
    ],
  ],
]
`;

exports[`codetalk complex.codetalk parses correctly 1`] = `
Array [
  Array [
    "codetalk",
    Array [
      "slide",
      Array [
        Array [
          Array [
            "#",
            "hello world",
          ],
          Array [],
        ],
        Array [
          Array [
            "##",
            "this is code talk",
          ],
          Array [
            Array [
              "
",
            ],
          ],
        ],
        Array [
          "idk why you say goodbye",
          Array [],
        ],
      ],
      "end",
    ],
  ],
]
`;

exports[`codetalk hello_world.codetalk parses correctly 1`] = `
Array [
  Array [
    "codetalk",
    Array [
      "slide",
      Array [
        Array [
          "hello world",
          Array [],
        ],
      ],
      "end",
    ],
  ],
]
`;
